docs(cypress): update to reflect new commands (#2833)
commercetools/merchant-center-application-kit: website/src/content/api-reference/commercetools-frontend-cypress.mdx

};
```

<Info>

> For Cypress version >= `10.0.0` the configuration changed. See [migration guide](https://docs.cypress.io/guides/references/migration-guide#Migrating-to-Cypress-version-10-0).

Reviewer: ```suggestion
The configuration has changed for Cypress version >= `10.0.0`. See the [migration guide](https://docs.cypress.io/guides/references/migration-guide#Migrating-to-Cypress-version-10-0).
```

> The task should be defined in the `cypress.config.ts` file.

Reviewer: ```suggestion
Define the task in the `cypress.config.ts` file.
```

```ts title="cypress.config.ts"
import { defineConfig } from 'cypress';
import { customApplicationConfig } from '@commercetools-frontend/cypress/task';

export default defineConfig({
  retries: 1,
  video: false,
  e2e: {
    async setupNodeEvents(on, cypressConfig) {
      on('task', {
        customApplicationConfig,
      });
      return cypressConfig
    },
    baseUrl: 'http://localhost:3001',
  },
});
```

</Info>

Extend the Cypress commands:

```js title="cypress/support/commands.js"
import '@commercetools-frontend/cypress/add-commands';
```

# Commands

## cy.loginToMerchantCenter

> This command performs the user login to the Merchant Center. It automatically detects whether the application is running on localhost or on production and chooses the appropriate login mechanism.<br/>

Reviewer: ```suggestion
This command performs the user login to the Merchant Center. It automatically detects whether the application is running on localhost or production and chooses the appropriate login mechanism.<br/>
```

> This command performs the user login to the Merchant Center. It automatically detects the whether the application is running on localhost or on production and chooses the appropriate login mechanism.<br/>

Reviewer 2: ```suggestion
This command performs the user login to the Merchant Center. It automatically detects whether the application is running on localhost or on production and chooses the appropriate login mechanism.<br/>
```

- When the application runs locally, the same mechanism used in the `cy.loginByOidc` is used.

> - When the application runs on production, a normal login flow is used where the user credentials are typed into the login form.

Reviewer: ```suggestion
- When the application runs on production, a normal login flow is used where the user enters the credentials into the login form.
```

### Usage

```ts
it('should render page 1', () => {
  cy.loginToMerchantCenter({
    entryPointUriPath: 'my-app',
    initialRoute: `/${Cypress.env('PROJECT_KEY')}/my-app/page-1`,
  });
})
it('should render page 2', () => {
  cy.loginToMerchantCenter({ entryPointUriPath: 'my-app' });
  cy.visit(`/${Cypress.env('PROJECT_KEY')}/my-app/page-2`)
})
```

### Options

Available options are:

> * `entryPointUriPath` (**required**): The application entry point URI path. This value is used to identify the correct application config.

Reviewer: ```suggestion
* `entryPointUriPath` (**required**): The application entry point URI path is used to identify the correct application config.
```

> * `dotfiles` (optional): Pass a list of dotenv files that must be loaded when the `custom-application-config.json` is loaded (in case you are using environment placeholder). By default the following dotenv files are loaded: `.env` and `.env.local`. You can also define the values using paths relative to the application folder.

Reviewer: ```suggestion
* `dotfiles` (optional): A list of dotenv files to load when the `custom-application-config.json` is loaded (in case you're using an environment placeholder). By default the following dotenv files are loaded: `.env` and `.env.local`. You can also define the values using paths relative to the application folder.
```

> * `initialRoute` (optional): If defined, the route is visited after login. If not defined, make sure to call `cy.visit` yourself.

Reviewer: ```suggestion
* `initialRoute` (optional): The route to open after login. If not defined, make sure to call `cy.visit` yourself.
```

> * `projectKey` (optional): The project key to access in the user session. The session token is valid for one project key at a time. Defaults to `Cypress.env('PROJECT_KEY')`.

Reviewer: ```suggestion
* `projectKey` (optional): The project key to access the user session. The session token is valid for one project key at a time. Defaults to `Cypress.env('PROJECT_KEY')`.
```

> * `onBeforeLoad` (optional): Function called before the page has loaded all of its resources. Use this as a chance to interact for example with the browser storage.

Reviewer: ```suggestion
* `onBeforeLoad` (optional): The function to call before the page has loaded all of its resources. Use this as a chance to interact, for example, with the browser storage.
```

* `login` (optional): An object with the user login credentials `email` and `password`. If not provided, the `email` defaults to `Cypress.env('LOGIN_EMAIL') || Cypress.env('LOGIN_USER')` and the `password` defaults to `Cypress.env('LOGIN_PASSWORD')`.
* `disableCacheAcrossSpecs` (optional): Turn off caching the session across specs. This is only relevant for Cypress version >= `10.9.0`.

> The command also requires to load the `custom-application-config.json` (automatically done via the Cypress task) and therefore it may need to load environment variables in case the application config uses environment placeholders.<br/>

Reviewer: ```suggestion
The command also requires loading the `custom-application-config.json` (automatically done via the Cypress task), so it may need to load environment variables in case the application config uses environment placeholders.<br/>
```

> By default, the `.env` and `.env.local` files are attempted to be loaded from the application folder. You can pass a `dotfiles` option to pass a list of names/paths relative to the application folder in case the files in the project have a different name/location.

Reviewer: ```suggestion
The `.env` and `.env.local` files are loaded by default from the application folder. You can pass a `dotfiles` option to provide a list of names/paths relative to the application folder in case the files in the project have a different name/location.
```

<Info>

> The command also requires the following environment variables to be available: `PROJECT_KEY`, `LOGIN_USER`, `LOGIN_PASSWORD`.

Reviewer 2: Since a user can provide the `login` option to this command, it seems `LOGIN_USER` AND `LOGIN_PASSWORD` env variables are not strictly required. Maybe they are needed internally for another thing?

Author: Yeah, let's rephrase that.

```suggestion
Make sure that the following environment variables are available in case the related options are not explicitly provided: `PROJECT_KEY`, `LOGIN_USER`, `LOGIN_PASSWORD`.
```

Author: @CarlosCortizasCT Something like that? ⬆️

Reviewer 2: Yes, I find it more precise.
Thanks 🙇

> Make sure that the following environment variables are available in case the related options are not explicitly provided: `PROJECT_KEY`, `LOGIN_USER`, `LOGIN_PASSWORD`.

Reviewer: ```suggestion
Ensure that the following environment variables are available if the related options aren't provided explicitly:  `PROJECT_KEY`, `LOGIN_USER`, `LOGIN_PASSWORD`.
```

</Info>

### Session

> The login command attempts to use the [cy.session](https://docs.cypress.io/api/commands/session) command, which caches and restores the user session between test runs.
> 
> This ultimately results in subsequent tests to run much faster (by restoring the previous session) and makes the test behave as if the user is already authenticated.
> 
> <Info>
> 
> Make sure to have the option `experimentalSessionAndOrigin` turned on (in your Cypress config) to enable that.

Reviewer: ```suggestion
The login command attempts to use the [cy.session](https://docs.cypress.io/api/commands/session) command, which caches and restores the user session between test runs.

This ultimately results in subsequent tests running much faster (by restoring the previous session) and making the test behave as if the user is authenticated.

<Info>

Make sure to have the option `experimentalSessionAndOrigin` turned on (in your Cypress config) to enable that.
```

</Info>

## cy.loginByOidc

<Info>

> This command is **deprecated**. We recommend to use the more generic `cy.loginToMerchantCenter` command as it automatically detects which login mechanism to use.

Reviewer: ```suggestion
This command is **deprecated**. We recommend using the more generic `cy.loginToMerchantCenter` command as it automatically detects which login mechanism to use.
```
